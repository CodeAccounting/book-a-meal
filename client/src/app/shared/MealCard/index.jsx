import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Dropdown from '../Dropdown';
import LinkBtn from '../Link';
import { mealObjPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function MealCard
 * @returns {JSX} MealCard
 */
const MealCard = ({ meal, type, toggleModal }) => (
  <div className="meal-card" id="meal-card">
    <div className="meal-card-header">
      <img src={meal.imageURL} alt="meal" />
      {type === 'caterer' &&
        <Dropdown
          type="card"
          toggler={<Fragment>&hellip;</Fragment>}
          content={
            <Fragment>
              <LinkBtn href="#add-edit-modal" id="edit-meal" clickHandler={() => toggleModal('editMeal')}>Edit</LinkBtn>
              <LinkBtn href="#add-edit-modal" id="delete-meal" clickHandler={() => toggleModal('deleteMeal')}>Delete</LinkBtn>
            </Fragment>
          }
        />}
      <div className="menu-card-title">
        <p>{meal.title}</p>
      </div>
    </div>
    <div className="meal-card-body">
      <div>
        <h3>&#8358; {meal.price}</h3>
        <p>{meal.description}</p>
      </div>
      {type === 'user' &&
        <div className="meal-card-action">
          <button className="btn btn-sec meal-card-btn">Click to Order</button>
        </div>}
    </div>
  </div>
);

MealCard.propTypes = {
  type: PropTypes.string.isRequired,
  meal: mealObjPropTypes.isRequired,
  toggleModal: PropTypes.func,
};

MealCard.defaultProps = {
  toggleModal: null
};

export default MealCard;
