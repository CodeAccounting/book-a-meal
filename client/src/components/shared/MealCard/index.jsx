import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Dropdown from '../Dropdown';
import LinkBtn from '../Link';
import { mealObjPropTypes } from '../../../helpers/proptypes';
import checkShopOpen from '../../../helpers/checkShopOpen';

/**
 * @exports
 * @function MealCard
 * @param {object} props
 * @returns {JSX} MealCard
 */
const MealCard = (props) => {
  const {
    meal, type, toggleModal, orderMeal
  } = props;

  const isShopOpen = checkShopOpen();

  return (
    <div className={`meal-card ${type === 'customer' && 'order-meal-card'}`} id="meal-card">
      <div className="meal-card-header">
        <img src={meal.imageURL} alt="meal" />
        {type === 'caterer' &&
          <Dropdown
            type="card"
            toggler={<Fragment>&hellip;</Fragment>}
            content={
              <Fragment>
                <LinkBtn href="#add-edit-modal" id="edit-meal" clickHandler={() => toggleModal('editMeal')}>Edit</LinkBtn>
                <LinkBtn href="#add-edit-modal" id="delete-meal" clickHandler={() => toggleModal('deleteMeal')}>Delete</LinkBtn>
              </Fragment>
            }
          />}
        <div className="menu-card-title">
          <p>{meal.title}</p>
        </div>
      </div>
      <div className="meal-card-body">
        <div>
          <h3>&#8358; {meal.price}</h3>
          <p>{meal.description}</p>
        </div>
        {type === 'customer' && isShopOpen &&
          <div className="meal-card-action">
            <button className="btn btn-sec meal-card-btn" onClick={orderMeal}>Click to Order</button>
          </div>}
      </div>
    </div>
  );
};

MealCard.propTypes = {
  type: PropTypes.string.isRequired,
  meal: mealObjPropTypes.isRequired,
  toggleModal: PropTypes.func,
  orderMeal: PropTypes.func,
};

MealCard.defaultProps = {
  toggleModal: null,
  orderMeal: null
};

export default MealCard;
