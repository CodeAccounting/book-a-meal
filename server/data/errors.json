{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}