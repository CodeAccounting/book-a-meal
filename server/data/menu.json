[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]