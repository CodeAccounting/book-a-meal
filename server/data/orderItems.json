[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]