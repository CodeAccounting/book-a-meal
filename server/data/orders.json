[
  {
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e544248c-145c-4145-b165-239658857637",
     "created": "2018-05-07T14:34:90.000",
     "updated": "2018-05-07T14:34:90.000",
     "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e5508b87-7975-493d-a900-3d47a69dad03",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  }
]