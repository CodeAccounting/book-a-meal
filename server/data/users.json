[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
    "username": null,
    "businessName": "FoodCircle",
    "email": "food@circle.com",
    "password": "foodcircle",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  },
  {
    "userId": "61bb8f8d-3b59-4294-acbc-16623818c391",
    "username": "Iveren",
    "businessName": null,
    "email": "iveren@shaguy.com",
    "password": "iverenshaguy",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": null,
    "businessAddress": null,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "user"
  },
  {
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "username": "Olisa",
     "businessName": null,
     "email": "olisa@emodi.com",
     "password": "olisaemodi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "username": "Emiola",
     "businessName": null,
     "email": "emiola@olasanmi.com",
     "password": "emiolaolasanmi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
    "username": null,
    "businessName": "BellyFill",
    "email": "belly@fill.com",
    "password": "bellyfil",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  }
]