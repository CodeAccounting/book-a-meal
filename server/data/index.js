import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
