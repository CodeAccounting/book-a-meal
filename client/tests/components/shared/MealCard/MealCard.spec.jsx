import React from 'react';
import MealCard from '../../../../src/app/shared/MealCard';
import { caterersMealsObj } from '../../../setup/data';

describe('MealModal', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when type is caterer', () => {
    const wrapper = shallow(<MealCard type="caterer" meal={caterersMealsObj.meals[0]} toggleModal={jest.fn()} />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('#edit-meal')).toBeTruthy();
  });

  it('renders correctly when type is user', () => {
    const wrapper = shallow(<MealCard type="user" meal={caterersMealsObj.meals[0]} toggleModal={jest.fn()} />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.meal-card-action')).toBeTruthy();
  });
});
