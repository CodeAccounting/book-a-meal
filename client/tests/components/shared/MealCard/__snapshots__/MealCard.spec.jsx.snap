// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealModal renders correctly when type is caterer 1`] = `
<div
  className="meal-card"
  id="meal-card"
>
  <div
    className="meal-card-header"
  >
    <img
      alt="meal"
      src="http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg"
    />
    <Dropdown
      content={
        <UNDEFINED>
          <Link
            className={null}
            clickHandler={[MockFunction]}
            href="#add-edit-modal"
            id="edit-meal"
          >
            Edit
          </Link>
          <Link
            className={null}
            clickHandler={null}
            href="#add-edit-modal"
            id="delete-meal"
          >
            Delete
          </Link>
        </UNDEFINED>
      }
      toggler={
        <UNDEFINED>
          …
        </UNDEFINED>
      }
      type="card"
    />
    <div
      className="menu-card-title"
    >
      <p>
        Jollof Rice, Beef and Plantain
      </p>
    </div>
  </div>
  <div
    className="meal-card-body"
  >
    <div>
      <h3>
        ₦ 
        1500.00
      </h3>
      <p>
        Jollof Rice, Beef and Plantain. 2 pieces of beef per plate
      </p>
    </div>
  </div>
</div>
`;

exports[`MealModal renders correctly when type is user 1`] = `
<div
  className="meal-card"
  id="meal-card"
>
  <div
    className="meal-card-header"
  >
    <img
      alt="meal"
      src="http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg"
    />
    <div
      className="menu-card-title"
    >
      <p>
        Jollof Rice, Beef and Plantain
      </p>
    </div>
  </div>
  <div
    className="meal-card-body"
  >
    <div>
      <h3>
        ₦ 
        1500.00
      </h3>
      <p>
        Jollof Rice, Beef and Plantain. 2 pieces of beef per plate
      </p>
    </div>
    <div
      className="meal-card-action"
    >
      <button
        className="btn btn-sec meal-card-btn"
      >
        Click to Order
      </button>
    </div>
  </div>
</div>
`;
